addons/account/models/partner.py

```diff
@@ -755,12 +755,16 @@ def _increase_rank(self, field, n=1):
         if self.ids and field in ['customer_rank', 'supplier_rank']:
             try:
                 with self.env.cr.savepoint(flush=False), mute_logger('odoo.sql_db'):
-                    query = sql.SQL("""
-                        SELECT {field} FROM res_partner WHERE ID IN %(partner_ids)s FOR NO KEY UPDATE NOWAIT;
-                        UPDATE res_partner SET {field} = {field} + %(n)s
+                    query = SQL("""
+                        SELECT %(field)s FROM res_partner WHERE ID IN %(partner_ids)s FOR NO KEY UPDATE NOWAIT;
+                        UPDATE res_partner SET %(field)s = %(field)s + %(n)s
                         WHERE id IN %(partner_ids)s
-                    """).format(field=sql.Identifier(field))
-                    self.env.cr.execute(query, {'partner_ids': tuple(self.ids), 'n': n})
+                        """,
+                        field=SQL.identifier(field),
+                        partner_ids=tuple(self.ids),
+                        n=n,
+                    )
+                    self.env.execute_query(query)
```

Comment:
maybe we could execute the query directly without using the variable but it will break the commit history, i let you choose if you prefer to be consistent with what it was before :smile:

Reply:
Not sure how it breaks commit history, but I merged it into a single statement.